sdk/storage/Azure.Storage.Blobs.Batching/src/Multipart.cs

```diff
@@ -0,0 +1,257 @@
+        /// messages.
+        /// </summary>
+        /// <param name="messages">
+        /// The batch sub-operation messages to submit together.
+        /// </param>
+        /// <param name="prefix">
+        /// A prefix used for the multipart/mixed boundary.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate notifications
+        /// that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A tuple containing the batch sub-operation messages merged into a
+        /// single multipart/mixed content stream and content type.
+        /// </returns>
+        public static Task<(Stream, string)> CreateAsync(
+            IEnumerable<HttpPipelineMessage> messages,
+            string prefix,
+#pragma warning disable CA1801 // Remove unused parameter (leaving for future changes)
+#pragma warning disable IDE0060 // Remove unused parameter
+            bool async,
+            CancellationToken cancellationToken)
+#pragma warning restore IDE0060 // Remove unused parameter
+#pragma warning restore CA1801 // Remove unused parameter
+        {
+            // Set the content-type
+            var boundary = $"{prefix}_{GetRandomGuid().ToString()}";
+            var contentType = BatchConstants.MultipartContentTypePrefix + boundary;
+
+            // TODO: Investigate whether to use a StreamWriter instead of a
+            // StringBuilder we turn into a MemoryStream
+            var content = new StringBuilder(1024);
+
+            const string newline = "\r\n";
+            var operationId = 0;
+            foreach (HttpPipelineMessage message in messages)
+            {
+                // Write the boundary
+                content.Append(BatchConstants.BatchSeparator).Append(boundary).Append(newline);
+                content.Append(BatchConstants.RequestContentType).Append(newline);
+                content.Append(BatchConstants.RequestContentTransferEncoding).Append(newline);
+                content.Append(BatchConstants.ContentIdName).Append(": ").Append(operationId++.ToString(CultureInfo.InvariantCulture)).Append(newline);
+                content.Append(newline);
+
+                // Write the request URI
+                content
+                    .Append(message.Request.Method.Method)
+                    .Append(" ")
+                    .Append(message.Request.Uri.PathAndQuery)
+                    .Append(" ")
+                    .Append(BatchConstants.HttpVersion)
+                    .Append(newline);
+
+                // Write the request headers
+                foreach (HttpHeader header in message.Request.Headers)
+                {
+                    content.Append(header.Name).Append(": ").Append(header.Value).Append(newline);
```

Comment:
Add to constants?

Reply:
I didn't think the space or colon were worth it...  if folks feel really strongly I can.